pkg/event/event.go

```diff
@@ -815,7 +815,7 @@ func (e *EventHandler) updateCLockClass(cfgName string, clkClass fbprotocol.Cloc
 		switch clkClass {
 		case fbprotocol.ClockClass6: // T-GM connected to a PRTC in locked mode (e.g., PRTC traceable to GNSS)
 			// update only when ClockClass is changed or clockAccuracy changes
-			if g.ClockQuality.ClockClass != fbprotocol.ClockClass6 {
+			if g.ClockQuality.ClockClass != fbprotocol.ClockClass6 || g.TimePropertiesDS.TimeTraceable != true {
```

Comment:
@josephdrichard make sure ClockClass is uninitialized every time  new profile is applied 
https://github.com/k8snetworkplumbingwg/linuxptp-daemon/blob/2f965ced1d036b877762fc1c3cb12afa82bc1fc6/pkg/event/event.go#L200